osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/infra/networking/OslpChannelHandlerServer.java

```diff
@@ -257,15 +276,18 @@ private int convertGpsCoordinateFromFloatToInt(final Float input) {
         return (int) (input * 1000000);
     }
 
-    private Oslp.Message handleConfirmRegisterDeviceRequest(final byte[] deviceId, final byte[] sequenceNumber,
+    private Oslp.Message handleConfirmRegisterDeviceRequest(final byte[] deviceUid, final byte[] sequenceNumber,
             final Oslp.ConfirmRegisterDeviceRequest confirmRegisterDeviceRequest) throws ProtocolAdapterException {
 
         try {
-            this.deviceRegistrationService.confirmRegisterDevice(deviceId,
+            this.deviceRegistrationService.confirmRegisterDevice(deviceUid,
                     SequenceNumberUtils.convertByteArrayToInteger(sequenceNumber),
                     confirmRegisterDeviceRequest.getRandomDevice(), confirmRegisterDeviceRequest.getRandomPlatform());
+
+            this.handleSetSchedule(deviceUid);
```

Comment:
Hmm, the pendingSetScheduleRequestService is transactional. If some other process deleted the pendingSetScheduleRequest this would indeed result in a fail. But I don't exactly know if table locks or record locks or whatever type of locking will be used I don't think it is a good idea to put all code within the handleSetSchedule in a transaction since this.deviceService.setSchedule(newDeviceRequest) will also be inside in the transaction. I think that operation will take most of the time, and I would prefer to at least have that operation succeed before the pendingsetschedulerequest record is removed. I'll change the service to catch PersistenceException.